### PR TITLE
ref(tsc): remove HTMLProps type intersection

### PR DESCRIPTION
**Pull request by JonasBa**
Small changes and slightly narrower types

### CODE REVIEW

File: static/app/components/hovercard.tsx [hunk 1]
```diff
@@ -127,10 +127,12 @@ function Hovercard(props: HovercardProps): React.ReactElement {
...
-  const hoverProps = useMemo((): Pick<
-    React.HTMLProps<HTMLDivElement>,
-    'onMouseEnter' | 'onMouseLeave'
-  > => {
+  const hoverProps = useMemo(():
+    | {
+        onMouseEnter: React.MouseEventHandler<HTMLDivElement>;
+        onMouseLeave: React.MouseEventHandler<HTMLDivElement>;
+      }
+    | {} => {
```
Comment (markstory) [inline static/app/components/hovercard.tsx:135]: What's going on this this union? Should these props be optional instead?

File: static/app/components/deprecatedforms/input.tsx [hunk 2]
```diff
@@ -2,11 +2,10 @@ import * as React from 'react';
...
 
-type Props = {
-  className?: string;
-} & React.HTMLProps<HTMLInputElement>;
-
-export default function Input({className, ...otherProps}: Props) {
+export default function Input({
+  className,
+  ...otherProps
+}: React.InputHTMLAttributes<HTMLInputElement>) {
```
Comment (evanpurkhiser) [inline static/app/components/deprecatedforms/input.tsx:8]: IMO it reads a little easier as 

```tsx
type Props = React.InputHTMLAttributes<HTMLInputElement>

export default function Input({className, ...otherProps}: Props) {
```

File: static/app/components/hovercard.tsx (hunk 1 above)
Comment (evanpurkhiser) [inline static/app/components/hovercard.tsx:135]: Why not just `Pick` seems clearer to me?

File: static/app/components/menuItem.tsx [hunk 3]
```diff
@@ -168,14 +170,14 @@ const MenuItem = ({
   );
 };
 
-type MenuListItemProps = {
+interface MenuListItemProps extends React.LiHTMLAttributes<HTMLLIElement> {
```
Comment (evanpurkhiser) [inline static/app/components/menuItem.tsx:173]: Wait what's the difference between this and `HTMLAttributes` 🤔

File: static/app/components/hovercard.tsx (hunk 1 above)
Comment (JonasBa) [reply static/app/components/hovercard.tsx:135]: Yeah, we can also make them optional. I probably went too strict on the type here

File: static/app/components/deprecatedforms/input.tsx (hunk 2 above)
Comment (JonasBa) [reply static/app/components/deprecatedforms/input.tsx:8]: Yeah, no preference tbh

File: static/app/components/hovercard.tsx (hunk 1 above)
Comment (JonasBa) [reply static/app/components/hovercard.tsx:135]: Pick achieves the same, but it's one level higher in the abstraction. Imo better to just pull the type directly so it's less ambiguous what Pick<HTMLAttributes<Element>["onClick"]>> is

File: static/app/components/menuItem.tsx (hunk 3 above)
Comment (JonasBa) [reply static/app/components/menuItem.tsx:173]: iirc this is just the react preset

**Review by markstory — APPROVED** [review]

**Pull request merged**